### PR TITLE
Update `_BatchManager` to work with `GlobalStep`s and `input_batch_size` per step

### PR DESCRIPTION
**Pull request by gabrielmbmb**
## Description

This PR updates the `_BatchManager` to handle the creation of a "super" batch gathering all the data from the previous batches for the `GlobalStep`s. In addition, it adds a new attribute to the `Step` class called `input_batch_size`. This argument can be used to define the desired number of rows that the step will receive per batch. Having that said, the `_BatchManager` has also been updated to store the data of the output batches, instead of the batches itself. This allows to build batches of the desired size for each step.

### CODE REVIEW

File: src/distilabel/utils/serialization_v2.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 SaveFormats = Literal["json", "yaml"]
 
 
-def _get_class(module: str = None, name: str = None) -> Type:
+def _get_class(module: str, name: str) -> Type:
```
Comment (plaguss) [inline src/distilabel/utils/serialization_v2.py:32]: The default args are placed so [here](https://github.com/argilla-io/distilabel/pull/366/files#diff-a566a0dfe56d6cf771f7e75367798fb6e01641133dce994665d5e256699a21a0R355) we can pass the data using **kwargs. But also, I see I have to update the code to use `distilabel/pipeline/serialization` and remove this file. Will do that in a different PR.

**Review by plaguss — COMMENTED** [review]
Just a comment irrelevant to the nice PR

File: src/distilabel/utils/serialization_v2.py (hunk 1 above)
Comment (gabrielmbmb) [reply src/distilabel/utils/serialization_v2.py:32]: oops, my bad!
